### PR TITLE
Upgrade IntelliJ Platform Gradle Plugin to 2.0 

### PR DESCRIPTION
**Pull request by rli**
Required for 2024.2 support

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -32,7 +32,7 @@ allprojects {
     configurations.all {
         resolutionStrategy {
             failOnDynamicVersions()
-            failOnChangingVersions()
+//            failOnChangingVersions()
```
Comment (manodnyab) [inline build.gradle.kts:35]: nit: do we need this or can be deleted?

File: plugins/amazonq/build.gradle.kts [hunk 2]
```diff
@@ -23,15 +25,11 @@ tasks.jar {
...
-    plugins.set(
-        listOf(
-            project(":plugin-core")
-        )
-    )
-}
-
 dependencies {
+    intellijPlatform {
+        localPlugin(project(":plugin-core", "pluginZip"))
```
Comment (manodnyab) [inline plugins/amazonq/build.gradle.kts:30]: what is the pluginZip used for?

File: build.gradle.kts (hunk 1 above)
Comment (rli) [reply build.gradle.kts:35]: ideally we keep this, not sure if we can yet

File: plugins/amazonq/build.gradle.kts (hunk 2 above)
Comment (rli) [reply plugins/amazonq/build.gradle.kts:30]: https://github.com/aws/aws-toolkit-jetbrains/pull/4524/files#diff-16d946c3e3b8d75616c236e9343f92d0dfb807ebbbcab7cf03037cdbc14693d7R90

**Review by manodnyab — APPROVED** [review]

File: buildspec/linuxTests.yml [hunk 3]
```diff
@@ -32,8 +32,12 @@ phases:
         fi
 
       - chmod +x gradlew
-      - su codebuild-user -c "./gradlew -PideProfileName=$ALTERNATIVE_IDE_PROFILE_NAME check coverageReport --info --console plain --continue"
-      - ./gradlew -PideProfileName=$ALTERNATIVE_IDE_PROFILE_NAME :plugin-toolkit:intellij-standalone:buildPlugin :plugin-core:buildPlugin :plugin-amazonq:buildPlugin
+      - su codebuild-user -c "./gradlew -PideProfileName=$ALTERNATIVE_IDE_PROFILE_NAME check coverageReport --info --console plain --continue -scan"
+      - ./gradlew -PideProfileName=$ALTERNATIVE_IDE_PROFILE_NAME :plugin-toolkit:intellij-standalone:buildPlugin :plugin-core:buildPlugin :plugin-amazonq:buildPlugin :plugin-toolkit:intellij-standalone:verifyPlugin :plugin-core:verifyPlugin :plugin-amazonq:verifyPlugin
```
[the remark this answers is not in the document]
Comment (rli) [reply buildspec/linuxTests.yml:36]: the new behavior appears to spin up multiple IDE instances and we run out of RAM in the build container

File: plugins/toolkit/jetbrains-gateway/tst/software/aws/toolkits/jetbrains/gateway/connection/SshCommandLineTest.kt [hunk 4]
```diff
@@ -63,7 +66,29 @@ class SshCommandLineTest {
         SshCommandLine("localhost", port = sshServer.server.port)
             .knownHostsLocation(tempFolder.newFile().toPath())
             .localPortForward(localPort, wireMockPort)
-            .executeInBackground()
+            .executeInBackground(
+                object : ProcessListener {
+                    override fun startNotified(event: ProcessEvent) {
+                        println("ssh client: startNotified")
```
[the remark this answers is not in the document]
Comment (rli) [reply plugins/toolkit/jetbrains-gateway/tst/software/aws/toolkits/jetbrains/gateway/connection/SshCommandLineTest.kt:72]: trying to understand the random failure in windows for this test

**Pull request merged**